utils/utils.go

```diff
@@ -0,0 +1,478 @@
+package utils
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	neturl "net/url"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"runtime"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/deis/workflow-e2e/shims"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
+)
+
+type Cmd struct {
+	Env               []string
+	CommandLineString string
+}
+
+type App struct {
+	Name string
+	URL  string
+}
+
+type Cert struct {
+	Name     string
+	CertPath string
+	KeyPath  string
+}
+
+const (
+	deisRouterServiceHost = "DEIS_ROUTER_SERVICE_HOST"
+	deisRouterServicePort = "DEIS_ROUTER_SERVICE_PORT"
+)
+
+var (
+	UuidRegExp  = `[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}`
+	ProcsRegexp = `(%s-v\d+-[\w-]+) up \(v\d+\)`
+
+	errMissingRouterHostEnvVar = fmt.Errorf("missing %s", deisRouterServiceHost)
+	DefaultMaxTimeout          = 5 * time.Minute // gomega's default is 2 minutes
+	Url                        = getController()
+	Debug                      = os.Getenv("DEBUG") != ""
+	HomeHome                   = os.Getenv("HOME")
+)
+
+func Cancel(url, username, password string) {
+	// log in to the account
+	Login(url, username, password)
+
+	// remove any existing test-* apps
+	sess, err := Run("deis apps")
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Exit(0))
+	re := regexp.MustCompile("test-.*")
+	for _, app := range re.FindAll(sess.Out.Contents(), -1) {
+		sess, err = Run("deis destroy --app=%s --confirm=%s", app, app)
+		Expect(err).To(BeNil())
+		Eventually(sess).Should(Say("Destroying %s...", app))
+		Eventually(sess).Should(Exit(0))
+	}
+
+	// cancel the account
+	sess, err = Run("deis auth:cancel --username=%s --password=%s --yes", username, password)
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Exit(0))
+	Eventually(sess).Should(Say("Account cancelled"))
+}
+
+func Login(url, user, password string) {
+	sess, err := Run("deis login %s --username=%s --password=%s", url, user, password)
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Exit(0))
+	Eventually(sess).Should(Say("Logged in as %s", user))
+}
+
+func Logout() {
+	sess, err := Run("deis auth:logout")
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Exit(0))
+	Eventually(sess).Should(Say("Logged out\n"))
+}
+
+func Run(cmdLine string, args ...interface{}) (*Session, error) {
```

Comment:
I get that this is now exported, but why rename this to `Run` (instead of `Start`)?